### PR TITLE
Add ColorType::Multiband enum variant to support samples>1 per pixel

### LINKED ISSUE #147: Support for arbitrary sample counts
For my use-case, I need to decode TIFFs which contain 20+ channels. Currently, image-tiff only supports specific `ColorType`s, although a previous (and slower) version of the crate did support `ColorType::Other`. I have an extremely messy patch at https://github.com/Masterchef365/image-tiff/tree/more_channels which works for my purposes, but other users of this crate might benefit from a more refined approach.

### LINKED ISSUE #214: Very minor changes to allow multiband GeoTiff reading
I realize that this has been proposed to various degrees before.  In the GIS space, geotiffs with multiple bands are relatively common.  It would be nice to be able to just read all of the band data into a vector. The coordinate projections can easily be handled manually since the relevant tags are already exposed.  There are only a couple of minor things blocking the read functionality:

- The GeoTiff is usually coming in as a Photometric interpretation of BlackIsZero (WhiteIsZero also is ok).  In **image.rs**, we'd need to allow more than 1 bits_per_sample to avoid the unsupported error. 
- Need to allow sample depth other than 1,3,4
- The decoding buffer size limit of 256mb is too small for many geotiffs, the unlimited settings work fine.

One way to implement the above would be to detect if a tiff is a geotiff (e.g. it has the coordinate project tag info present probably works) and then just bypass the above restrictions just for geotiffs w/o impacting any of the current functionality for non-geotiffs  I was able to do this quickly and the existing ColorType::Gray ends up working fine for the different geotiff band data types.  This isn't going work for all possible geotiffs (e.g bands with different data types won't work currently), but that's a less common use case (and commented as a todo in the code already, so maybe it happens someday).  There are a couple of existing geotiff crates, but they were built for a single use case and appear to be largely unused/unsupported.  I am happy to code up the changes (all 5 lines) and submit, but wanted to check first if a line had already been drawn on this topic and it isn't going to happen.

### PR DESCRIPTION
**Pull request by weiji14**
Building on the work at #216 to support reading multi-band TIFF images (common in GeoTIFFs).

This PR extends the work done by @tromper in #216 which added the unit test for reading a 5-band GeoTIFF. I've just added the `ColorType::Multiband` enum variant as suggested at https://github.com/image-rs/image-tiff/pull/216#discussion_r1506951296), and made sure that the decoder functions can handle this multi-band type.

Refernces:
- https://www.awaresystems.be/imaging/tiff/tifftags/samplesperpixel.html

Supersedes #216, closes #214 and resolves #147.

### CODE REVIEW

File: src/decoder/image.rs [hunk 1]
```diff
@@ -347,14 +347,19 @@ impl Image {
...
-                if self.samples == 1 =>
-            {
-                Ok(ColorType::Gray(self.bits_per_sample))
+            PhotometricInterpretation::BlackIsZero | PhotometricInterpretation::WhiteIsZero => {
+                match self.samples {
+                    1 => Ok(ColorType::Gray(self.bits_per_sample)),
+                    _ => Ok(ColorType::Multiband {
+                        bit_depth: self.bits_per_sample,
+                        num_samples: self.samples,
+                    }),
```
Comment (weiji14) [inline src/decoder/image.rs:356]: Have kept the 1-band image as `ColorType::Gray` to preserve backwards compatibility as mentioned at https://github.com/image-rs/image-tiff/pull/216#discussion_r1506951296. TIFF files with bands>1 would use the new `ColorType::Multiband` enum variant here.

File: src/decoder/image.rs [hunk 2]
```diff
@@ -562,18 +567,25 @@ impl Image {
...
                     ));
                 }
             },
+            (
+                ColorType::Multiband {
+                    bit_depth: _n,
+                    num_samples: _samples,
+                },
+                _,
+            ) => {}
```
Comment (weiji14) [inline src/decoder/image.rs:584]: A little unsure if this should be a separate branch, or go in one of the branches above. Not familiar with what the Horizontal and FloatingPoint predictors are.

Comment (fintelia) [reply src/decoder/image.rs:584]: The predictors are filters that are applied to the pixel data before compression, to help improve the compression ratio. Multiband should be included in both the match existing match arms where Gray is.

Comment (weiji14) [reply src/decoder/image.rs]: Ok, done at e0b9ae040fe08842edc8f58182f89a31aa0c0338, but best if you can double check it. I don't have an multi-band TIFF file with a Horizontal/FloatingPoint predictor to test this out on.

**Pull request merged**